src/inflators/model.tsx

```diff
@@ -1,30 +1,82 @@
-import { addComponent, addEntity } from "bitecs";
-import { Object3D } from "three";
+import { addComponent, addEntity, hasComponent } from "bitecs";
+import { Material, Object3D } from "three";
 import { HubsWorld } from "../app";
-import { GLTFModel } from "../bit-components";
-import { addObject3DComponent, gltfInflatorExists, gltfInflators } from "../utils/jsx-entity";
+import { GLTFModel, MaterialTag } from "../bit-components";
+import { addMaterialComponent, addObject3DComponent, gltfInflatorExists, gltfInflators } from "../utils/jsx-entity";
+import { mapMaterials } from "../utils/material-utils";
+import { EntityID } from "../utils/networking-types";
 
 function camelCase(s: string) {
   return s.replace(/-(\w)/g, (_, m) => m.toUpperCase());
 }
 
 export type ModelParams = { model: Object3D };
 
+// These components are all handled in some special way, not through inflators
+const ignoredComponents = ["visible", "frustum", "frustrum", "shadow", "networked"];
+
+function inflateComponents(
```

Comment:
Despite the trickiness/book-keeping for refs, I like how simple this function is: For each non-ignored component, resolve its refs and call its inflator.